#! /bin/sh -e

# DP: capi20/capi20.c: Fix typo.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- capi20/capi20.c~	2004-12-15 15:27:54.000000000 +0100
+++ capi20/capi20.c	2004-12-25 15:53:52.000000000 +0100
@@ -492,7 +492,7 @@
     fd = applid2fd(ApplID);
 
     if ((*Buf = rcvbuf = get_buffer(ApplID, &bufsiz, &offset)) == 0)
-        return CapiMsgOSResourceError;
+        return CapiMsgOSResourceErr;
 
     if ((rc = read(fd, rcvbuf, bufsiz)) > 0) {
 	CAPIMSG_SETAPPID(rcvbuf, ApplID); // workaround for old driver
